src/modules/rppt_tensor_geometric_augmentations.cpp

```diff
@@ -1269,4 +1394,84 @@ return RPP_SUCCESS;
 #endif // backend
 }
 
+/******************** rotate ********************/
+
+RppStatus rppt_rotate_gpu(RppPtr_t srcPtr,
+                          RpptDescPtr srcDescPtr,
+                          RppPtr_t dstPtr,
+                          RpptDescPtr dstDescPtr,
+                          Rpp32f *angle,
+                          RpptInterpolationType interpolationType,
+                          RpptROIPtr roiTensorPtrSrc,
+                          RpptRoiType roiType,
+                          rppHandle_t rppHandle)
+{
+#ifdef HIP_COMPILE
+    if ((interpolationType != RpptInterpolationType::BILINEAR) && (interpolationType != RpptInterpolationType::NEAREST_NEIGHBOR))
+        return RPP_ERROR_NOT_IMPLEMENTED;
+    
+    // Compute affine transformation matrix from rotate angle
+    Rpp32f *affineTensor = rpp::deref(rppHandle).GetInitHandle()->mem.mcpu.tempFloatmem;
```

Comment:
why is this stored in cpu mem for HIP backend?

Reply:
The hip_exec_warp_affine_tensor internally does a hipMemcopy [here ](https://github.com/GPUOpen-ProfessionalCompute-Libraries/rpp/blob/a364d41a299b4d017c080dce7ed5233aed7ca9a4/src/modules/hip/kernel/warp_affine.hpp#L338)for the affine tensor buffer.
Hence we are computing the affine tensor and storing it in a pre existing memory in the handle and reusing the existing warp affine kernels for Rotate augmentation.